src/main/java/org/openmaptiles/layers/WaterName.java

```diff
@@ -214,38 +218,63 @@ public void process(Tables.OsmMarinePoint element, FeatureCollector features) {
   @Override
   public void process(Tables.OsmWaterPolygon element, FeatureCollector features) {
     if (nullIfEmpty(element.name()) != null) {
-      Geometry centerlineGeometry = lakeCenterlines.get(element.source().id());
-      FeatureCollector.Feature feature;
-      int minzoom = 9;
-      String place = element.place();
-      String clazz;
-      if ("bay".equals(element.natural())) {
-        clazz = FieldValues.CLASS_BAY;
-      } else if ("sea".equals(place)) {
-        clazz = FieldValues.CLASS_SEA;
-      } else {
-        clazz = FieldValues.CLASS_LAKE;
-        minzoom = 3;
-      }
-      if (centerlineGeometry != null) {
-        // prefer lake centerline if it exists
-        feature = features.geometry(LAYER_NAME, centerlineGeometry)
-          .setMinPixelSizeBelowZoom(13, 6d * element.name().length());
-      } else {
-        // otherwise just use a label point inside the lake
-        feature = features.pointOnSurface(LAYER_NAME)
-          .setMinZoom(place != null && SEA_OR_OCEAN_PLACE.contains(place) ? 0 : 3)
-          .setMinPixelSize(128); // tiles are 256x256, so 128x128 is 1/4 of a tile
+      try {
+        Geometry centerlineGeometry = lakeCenterlines.get(element.source().id());
+        int minzoomCL = MINZOOM_BAY;
+        String place = element.place();
+        String clazz;
+        if ("bay".equals(element.natural())) {
+          clazz = FieldValues.CLASS_BAY;
+        } else if ("sea".equals(place)) {
+          clazz = FieldValues.CLASS_SEA;
+        } else {
+          clazz = FieldValues.CLASS_LAKE;
+          minzoomCL = MINZOOM_LAKE;
+        }
+        if (centerlineGeometry != null) {
+          // prefer lake centerline if it exists, but point will be also used if minzoom bellow 9 is calculated from area
+          // note: Here we're diverging from OpenMapTiles: For bays with minzoom (based on area) point is used between
+          // minzoom and Z8 and for Z9+ centerline is used, while OpenMaptiles sticks with points.
```

Comment:
why? Is planetiler-openmaptiles handling considered "better" here?

Reply:
Bay name is rendered along the centerline, instead of as usual label used for points.